pkg/reconciler/labeler/v2/accessors.go

```diff
@@ -57,9 +59,37 @@ func NewRevisionAccessor(
 	}
 }
 
-// get implements Accessor
-func (r *Revision) get(ns, name string) (kmeta.Accessor, error) {
-	return r.revisionLister.Revisions(ns).Get(name)
+// makeMetadataPatch makes a metadata map to be patched or nil if no changes are needed.
+func makeMetadataPatch(acc kmeta.Accessor, routeName *string) (map[string]interface{}, error) {
+	labels := map[string]interface{}{}
+
+	if err := addRouteLabel(acc, labels, routeName); err != nil {
+		return nil, err
+	}
+
+	if len(labels) > 0 {
+		meta := map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": labels,
+			},
+		}
+		return meta, nil
+	}
+	return nil, nil
+}
+
+// addRouteLabel appends the route label to the list of labels if needed.
```

Comment:
Do we always append? Since if routeName == nil and we do nothing the code can be simplified a bit.

Reply:
Updated the comment - if routeName is nil we actually append nil (which tells mergepatch to remove the value)